core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -291,6 +293,26 @@ public <T extends ConnectionDetails> boolean test( T connectionDetails ) throws
     return connectionProvider.test( connectionDetails );
   }
 
+  /**
+   * Tests if a given VFS connection is valid, optionally, with certain testing options.
+   *
+   * @param connectionDetails The VFS connection.
+   * @param options The testing options, or {@code null}.
+   * @return {@code true} if the connection is valid; {@code false}, otherwise.
+   */
+  public <T extends VFSConnectionDetails> boolean test( T connectionDetails, VFSConnectionTestOptions options )
+          throws KettleException {
+    if ( Objects.isNull( connectionDetails ) ) {
+      throw new IllegalArgumentException( "Connection Details can't be null." );
+    }
+    VFSConnectionProvider<T> connectionProvider = (VFSConnectionProvider<T>) connectionProviders.get(connectionDetails.getType());
+
+    if ( Objects.isNull( options ) ) {
+      return connectionProvider.test( connectionDetails, new VFSConnectionTestOptions() );
+    }
+    return connectionProvider.test( connectionDetails, options );
```

Comment:
I think it would be better this way:
```java
    if ( options == null ) {
      options = new VFSConnectionTestOptions();
    }

    return connectionProvider.test( connectionDetails, options );
```

Reply:
addressed.